plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7623,6 +7625,75 @@ public void testSystemTables()
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testReplaceTableExtraProperties()
+    {
+        String tableName = "replace_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+        assertUpdate("CREATE OR REPLACE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.three'], ARRAY['three']))".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.three')".formatted(tableName),
+                "VALUES 'three'");
+    }
+
+    @Test
+    public void testExtraPropertiesWithCtas()
+    {
+        String tableName = "create_table_ctas_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+         assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two')".formatted(tableName),
+                "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testShowCreateWithExtraProperties()
+    {
+        String tableName = format("%s.%s.show_create_table_with_extra_properties_%s", getSession().getCatalog().get(), getSession().getSchema().get(), randomNameSuffix());
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        // Assert that SHOW CREATE TABLE does not contain extra_properties
+        assertThat((String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue())
+                .satisfies(new Condition<>(
+                        queryResult -> queryResult.contains("extra_properties"), "noExtraProperties"
+                ));
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testDuplicateExtraProperties()
+    {
+        assertQueryFails(
+                "CREATE TABLE create_table_with_duplicate_extra_properties (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property', 'extra.property'], ARRAY['true', 'false']))",
+                "Invalid value for catalog 'iceberg' table property 'extra_properties': Cannot convert.*");
+    }
+
+    @Test
+    public void testNullExtraProperty()
+    {
+        assertQueryFails(
+                "CREATE TABLE create_table_with_duplicate_extra_properties (c1 integer) WITH (extra_properties = MAP(ARRAY['null.property'], ARRAY[null]))",
+                ".*Extra table property value cannot be null '\\{null.property=null}'.*");
+    }
+
```

Comment:
Add a test to ensure the statement fails (both CREATE TABLE and CREATE TABLE AS SELECT ) when the extra_properties contains a reserved Trino IcebergTableProperties name.